CMakeLists.txt

```diff
@@ -5,9 +5,19 @@
 cmake_minimum_required(VERSION 3.0)
 project("Libmultiprocess" CXX)
 include(CTest)
+include(CheckCXXSourceCompiles)
 find_package(Boost)
 find_package(CapnProto)
 
+check_cxx_source_compiles("
```

Comment:
This check was failing for me due to a link error (linux requires pthreads programs to be linked with a pthreads library), so I wasn't seeing thread names after this. Following fix worked for me:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16ecf50..cf87baf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,11 +4,15 @@
 
 cmake_minimum_required(VERSION 3.0)
 project("Libmultiprocess" CXX)
+include(CMakePushCheckState)
 include(CTest)
 include(CheckCXXSourceCompiles)
 find_package(Boost)
 find_package(CapnProto)
+find_package(Threads REQUIRED)
 
+cmake_push_check_state()
+set(CMAKE_REQUIRED_LIBRARIES Threads::Threads)
 check_cxx_source_compiles("
   #include <pthread.h>
   int main(int argc, char** argv)
@@ -17,6 +21,7 @@ check_cxx_source_compiles("
     return pthread_getname_np(pthread_self(), thread_name, sizeof(thread_name));
   }"
   HAVE_PTHREAD_GETNAME_NP)
+cmake_pop_check_state()
 
 capnp_generate_cpp(MP_PROXY_SRCS MP_PROXY_HDRS include/mp/proxy.capnp)
 
```

Reply:
Excellent, done!